File: test/end-to-end/Classes/declarations.ts
-
class Point {
    public x: number;
    public y: number;
    protected square: number;
    private name: string;
    
    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
        this.square = x * y;
        this.name = "";
    }
}

class Measurements<T> {
    constructor(items: T[]) {
        // ...
    }
}

class Shape extends Measurements<Point> {
    constructor(points: Point[]) {
        // ...
    }
}
-

File: test/end-to-end/Classes/declarations.py
-
class Point:
    
    def __init__(self, x, y):
        self.x = x
        self.y = y
        self._square = x * y
        self.__name = ""

class Measurements:
    def __init__(self, items):
        # ...

class Shape(Measurements):
    def __init__(self, points):
        # ...
-
